content/blog/taking-control-of-plot-scaling/index.Rmd

```diff
@@ -0,0 +1,182 @@
+knitr::include_graphics(pngfile)
+```
+
+We can see that we are getting there, but the journey hasn't been pleasant. Especially for the last part it requires knowledge of all the different settings in a geom that encodes absolute sizes. For the mark geom we only fixed the ellipse line width and the text size, but there are many more settings that needs to be updated as well as is apparent from the weird look of the text box.
+
+Another issue that arises is that if we need this plot at yet another different scale, we will need to change quite a lot of code in order to get there.
+
+## Attempt 2: Resolution scaling
+Since the resolution is the parameter that ties the pixel size and absolute size together it is possible to use it as a scaling factor, but it requires some non-obvious adjustments:
+
+The first thing we need to do is convert our physical dimensions to pixel dimensions using our desired resolution. We want to end up with a 60x36cm plot at 300ppi. This gives us:
+
+```{r, message=FALSE, out.width=""}
+c(60, 36) *
+  0.3937 * # convert to inch
+  300 # convert to pixels
+```
+
+We can now use these values directly in our device and change the resolution of the device to trick it into thinking that text etc should be rendered at a larger size
+
+```{r, message=FALSE, out.width=""}
+pngfile <- fs::path(knitr::fig_path(),  "resolution.png")
+agg_png(pngfile, width = 7087, height = 4252, units = "px", res = 900)
+plot(p)
+invisible(dev.off())
+knitr::include_graphics(pngfile)
+```
+
+This actually works exactly as we hoped. We've gotten our huge version of the plot but with the same exact scaling of all graphic elements.
+
+Depending on your temperament this may be a perfect solution. To me, I think the conversion from physical dimensions to pixels is tedious, and it has the added drawback that the dimensions of the plot is incorrectly encoded (it will appear as a 20*12cm plot at 900ppi) which may impact how it is presented in different programs. In the end you may have to manually resize it to get the correct physical dimensions in the end.
+
+## The solution
+Seeing that there is no single perfect solution to fixing this with the tools at our disposal, I've added a new argument to the ragg devices called `scaling`. It's a multiplier that is applied to all absolute sizes, without interfering with the encoded dimensions of the output. Since we have increased the dimensions 3 times we set `scaling = 3` to make sure that the absolute sized elements are keeping their relative size.
+
+```{r, message=FALSE, out.width=""}
+pngfile <- fs::path(knitr::fig_path(),  "scaling.png")
+agg_png(pngfile, width = 60, height = 36, units = "cm", res = 300, scaling = 3)
+plot(p)
+invisible(dev.off())
+knitr::include_graphics(pngfile)
+```
+
+As can be seen, the new argument makes it very easy to reclaim the look of the plot after resizing. Hopefully this will remove a good deal of the pain related to generating plots for papers, posters, presentations, etc.
```

Comment:
Maybe mention that if you have separate plots, you'll need to set the `scaling` to the same value everywhere to ensure that plot sizes are consistent. And probably worth connecting back to beginning to remind people where that 3 comes from, and how you should adjust it for other use cases.